### PR TITLE
fix(seach): Ensure an `aggregated_def` is not a user-created tag

### PR DESCRIPTION
**Pull request by lauryndbrown**
If a user has a custom tag name `priority` and entered `priority:high` in the search box, the result would be a `500`. This is because `priority` is also the name of a possible aggregation that could clash with this. This PR adds a check to ensure that a user-created tag is not treated as a reserved word for an aggregation.

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -251,7 +251,8 @@ def visit_raw_search(self, node, children):
             SearchValue(value),
         )
 
-    def visit_numeric_filter(self, node, (search_key, _, operator, search_value)):
+    def visit_numeric_filter(self, node, xxx_todo_changeme):
```
Comment (wedamija) [inline src/sentry/api/event_search.py]: What are these changes?

File: tests/snuba/search/test_backend.py [hunk 2]
```diff
@@ -472,6 +472,62 @@ def test_bookmarked_by_with_environment(self):
...
+                project_id=self.project.id,
+            )
+        results = self.make_query(
+            search_filter_query='priority:%s' % priority,
+            tags={'priority': priority},
+            sort_by='priority',
+        )
+        # HELP(lb): I dont think this is testing what I want it to.
+        # I just want to ensure that the priority sort still works with the priority tag.
+        assert set(results) == set([self.group2])
```
Comment (wedamija) [inline tests/snuba/search/test_backend.py]: Seems ok to me. Might even be worth just combining this into the previous test to reduce duplication

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (lauryndbrown) [reply src/sentry/api/event_search.py]: I think it was just my linter. I can discard those changes

File: tests/snuba/search/test_backend.py (hunk 2 above)
Comment (lauryndbrown) [reply tests/snuba/search/test_backend.py]: Cool but how do you know that it was actually sorted? :/

Comment (lauryndbrown) [reply tests/snuba/search/test_backend.py]: I want to test that I did not break anything. I figure I didn't but I want to write a test in case.

Comment (wedamija) [reply tests/snuba/search/test_backend.py]: In that case, just make another group that has this tag, then make sure that one group will be before the other based on `aggregation_defs['priority']`. You'll just need to create an event that has a different fingerprint than the current one.

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/event_search.py]: Looks like it happened in a few methods.

**Comment by lauryndbrown** [comment]
@markstory and @wedamija could you let me know what you think now?

File: tests/snuba/search/test_backend.py [hunk 3]
```diff
@@ -472,6 +472,83 @@ def test_bookmarked_by_with_environment(self):
...
+            },
+            project_id=self.project.id,
+        )
+
+        self.group1.times_seen = 6
+        self.group2.times_seen = 3
+        self.group1.save()
+        self.group2.save()
+
+        group1_score = Group.calculate_score(self.group1.times_seen, self.group1.last_seen)
```
Comment (wedamija) [inline tests/snuba/search/test_backend.py]: TIL `Group.calculate_score`

File: tests/snuba/search/test_backend.py [hunk 4]
```diff
@@ -472,6 +472,83 @@ def test_bookmarked_by_with_environment(self):
...
+        group2_score = Group.calculate_score(self.group2.times_seen, self.group2.last_seen)
+        assert group1_score > group2_score
+
+        results = self.make_query(
+            search_filter_query='priority:%s' % priority,
+            tags={'priority': priority},
+            sort_by='priority',
+        )
+        assert set(results) == set([self.group1, self.group2])
```
Comment (wedamija) [inline tests/snuba/search/test_backend.py]: This doesn't really test the ordering since you're using a set. I'd just compare lists in the expected order.

**Review by wedamija — APPROVED** [review]

File: tests/snuba/search/test_backend.py (hunk 4 above)
Comment (lauryndbrown) [reply tests/snuba/search/test_backend.py]: oops! you're totally right! Thanks for the catch!

File: tests/snuba/search/test_backend.py [hunk 5]
```diff
@@ -472,6 +472,74 @@ def test_bookmarked_by_with_environment(self):
...
+        results = self.make_query(
+            search_filter_query='priority:%s' % priority,
+            tags={'priority': priority},
+        )
+
+        assert set(results) == set([self.group2])
+
+    def test_search_filter_query_with_custom_priority_tag_and_priority_sort(self):
+        priority = 'high'
+        for i in range(1, 10):
```
Comment (wedamija) [inline tests/snuba/search/test_backend.py]: I'd probably just make 2-3 events typically if that works for the sort, just to save on making the test slow

**Review by wedamija — APPROVED** [review]

**Pull request merged**